scripts/release/debian/deb_install.sh

```diff
@@ -81,13 +84,22 @@ setup() {
             exit 1
         fi
     fi
-    echo "deb [arch=$(dpkg --print-architecture)] https://packages.microsoft.com/repos/azure-cli/ ${CLI_REPO} main" \
-        > /etc/apt/sources.list.d/azure-cli.list
+
+    if [ -f /etc/apt/sources.list.d/azure-cli.list ]; then
+      rm /etc/apt/sources.list.d/azure-cli.list
+    fi
+
+    echo "Types: deb
+URIs: https://packages.microsoft.com/repos/azure-cli/
+Suites: ${CLI_REPO}
+Components: main
+Architectures: $(dpkg --print-architecture)
+Signed-by: /etc/apt/keyrings/microsoft.gpg" | tee /etc/apt/sources.list.d/azure-cli.sources
```

Comment:
This `/etc/apt/sources.list.d/azure-cli.list` -> `/etc/apt/sources.list.d/azure-cli.sources` renaming breaks the logic of `check_distro_consistency` from https://github.com/Azure/azure-cli/pull/5755:

https://github.com/Azure/azure-cli/blob/b6ab754b68efba60b30806fa80dc3d0377be84ca/src/azure-cli-core/azure/cli/core/extension/operations.py#L44

https://github.com/Azure/azure-cli/blob/b6ab754b68efba60b30806fa80dc3d0377be84ca/src/azure-cli-core/azure/cli/core/extension/operations.py#L564-L591

and result in error in debug log:

```
cli.azure.cli.core.extension.operations: [Errno 2] No such file or directory: '/etc/apt/sources.list.d/azure-cli.list'
```